src/OrchardCore.Modules/OrchardCore.Email.Smtp/Drivers/SmtpSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,113 @@
+        _smtpEmailSettings = smtpEmailSettings.Value;
+    }
+
+    public override async Task<IDisplayResult> EditAsync(SmtpEmailSettings settings, BuildEditorContext context)
+    {
+        var user = _httpContextAccessor.HttpContext?.User;
+
+        if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageSmtpEmailSettings))
+        {
+            return null;
+        }
+
+        var shapes = new List<IDisplayResult>
+        {
+            Initialize<SmtpEmailSettings>("SmtpEmailSettings_Edit", model =>
+            {
+                model.DefaultSender = _smtpEmailSettings.DefaultSender;
+                model.DeliveryMethod = settings.DeliveryMethod;
+                model.PickupDirectoryLocation = settings.PickupDirectoryLocation;
+                model.Host = settings.Host;
+                model.Port = settings.Port;
+                model.ProxyHost = settings.ProxyHost;
+                model.ProxyPort = settings.ProxyPort;
+                model.EncryptionMethod = settings.EncryptionMethod;
+                model.AutoSelectEncryption = settings.AutoSelectEncryption;
+                model.RequireCredentials = settings.RequireCredentials;
+                model.UseDefaultCredentials = settings.UseDefaultCredentials;
+                model.UserName = settings.UserName;
+                model.Password = settings.Password;
+                model.IgnoreInvalidSslCertificate = settings.IgnoreInvalidSslCertificate;
+            }).Location("Content:5").OnGroup(GroupId),
+        };
+
+        if (_smtpEmailSettings.DefaultSender != null)
+        {
+            shapes.Add(Dynamic("SmtpEmailSettings_TestButton").Location("Actions").OnGroup(GroupId));
+        }
+
+        return Combine(shapes);
+    }
+
+    public override async Task<IDisplayResult> UpdateAsync(SmtpEmailSettings section, BuildEditorContext context)
+    {
+        var user = _httpContextAccessor.HttpContext?.User;
+
+        if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageSmtpEmailSettings))
+        {
+            return null;
+        }
+
+        if (context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase))
+        {
+            await context.Updater.TryUpdateModelAsync(section, Prefix);
+
+            var previousPassword = section.Password;
+            // Restore password if the input is empty, meaning that it has not been reset.
+            if (string.IsNullOrWhiteSpace(section.Password))
+            {
+                section.Password = previousPassword;
+            }
+            else
+            {
+                // Encrypt the password.
+                var protector = _dataProtectionProvider.CreateProtector(nameof(SmtpEmailSettingsConfiguration));
+                section.Password = protector.Protect(section.Password);
+            }
+
+            // Release the tenant to apply the settings.
+            await _shellHost.ReleaseShellContextAsync(_shellSettings);
```

Comment:
This is how it was

Reply:
Okay, can you change it? We should not be releasing just because the user clicked save without making an actual change. Releasing the shell recreate the tenant all over again which is a very expensive operation.

If you don't want to change it here, we'll probably need to revisit this and change it.